Index: linux-4.4.178/include/uapi/linux/if_ether.h
===================================================================
--- linux-4.4.178.orig/include/uapi/linux/if_ether.h
+++ linux-4.4.178/include/uapi/linux/if_ether.h
@@ -22,6 +22,7 @@
 #define _UAPI_LINUX_IF_ETHER_H
 
 #include <linux/types.h>
+#include <linux/libc-compat.h>
 
 /*
  *	IEEE 802.3 Ethernet magic constants.  The frame sizes omit the preamble
@@ -137,7 +138,10 @@
  */
 
 /* allow libcs like musl to deactivate this, glibc does not implement this. */
-#ifndef __UAPI_DEF_ETHHDR
+
+#if defined(_NETINET_IF_ETHER_H)
+#define __UAPI_DEF_ETHHDR		0
+#else
 #define __UAPI_DEF_ETHHDR		1
 #endif
 
Index: linux-4.4.178/include/uapi/linux/libc-compat.h
===================================================================
--- linux-4.4.178.orig/include/uapi/linux/libc-compat.h
+++ linux-4.4.178/include/uapi/linux/libc-compat.h
@@ -160,6 +160,8 @@
 #define __UAPI_DEF_IF_NET_DEVICE_FLAGS 1
 /* For the future if glibc adds IFF_LOWER_UP, IFF_DORMANT and IFF_ECHO */
 #define __UAPI_DEF_IF_NET_DEVICE_FLAGS_LOWER_UP_DORMANT_ECHO 1
+/* Definitions for if_ether.h */
+#define __UAPI_DEF_ETHHDR              1
 
 /* Definitions for in.h */
 #define __UAPI_DEF_IN_ADDR		1
